src/lib/supabase/library/library.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  mockGetAllLibraryItems,
+  mockSearchLibraryItemsByAvailability,
+  mockSearchLibraryItemsByCategory,
+  mockSearchLibraryItemsByName,
+  mockSearchLibraryItemsByTags,
+  mockSearchAllLibraryItems,
+} from './mockData';
+import {
+  LibraryItem,
+  Hardware,
+  Book,
+  Film,
+  Controller,
+  Accessory,
+} from '~/models/models';
```

Comment:
I'd alphabetize these imports to keep it readable. There's a lot of repeated types in these methods, you could consolidate them by creating a type at the top of the page, or alternatively in the `models` folder where those other items are stored. 
It would look something like this:

```suggestion
import {
  Accessory,
  Book,
  Controller,
  Film,
  Hardware,
  LibraryItem,
} from '~/models/models';

type TLibraryItems =
  | Accessory
  | Book
  | Controller
  | Film
  | Hardware
  | LibraryItem;

```

BTW, not sure if you know this or not, but you can commit my suggestion by just pressing the "commit this suggestion" button (or whatever its called) makes it so you don't have to copy / paste stuff or re-write in vs code.